test/state/stateV2_v3_migration.test.ts

```diff
@@ -0,0 +1,131 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { publishState } from "../utils/state-utils";
+import { DeployHelper } from "../../helpers/DeployHelper";
+import bigInt from "big-integer";
+import { StateContractMigrationHelper } from "../../helpers/StateContractMigrationHelper";
+import { NetworkIdTypes } from "../../helpers/NetworkIdTypes";
+
+const stateTransitionsWithProofs = [
+    require("./data/user_state_genesis_transition.json"),
+    require("./data/user_state_next_transition.json"),
+  ];
+
+  const stateTransitionsWithNoProofs = [
+    {
+      id: '6901746346790563787434755862277025452451108972170386555162524223864832',
+      oldState: '1099511627776',
+      newState: '2199023255552',
+      isOldStateGenesis: true,
+    },
+    {
+      id: '6901746346790563787434755862277025452451108972170386555162524223864832',
+      oldState: '2199023255552',
+      newState: '3298534883328',
+      isOldStateGenesis: false,
+    },
+  ];
+
+describe.only("Deploy StateV2 v2 and migrate", () => {
+    let state;
+    let guWrpr;
+    let deployHelper;
+    let signers;
+    const defaultIdType = NetworkIdTypes.polygonMumbai;
+
+    before(async function () {
+        signers = await ethers.getSigners();
+        deployHelper = await DeployHelper.initialize();
+        const contracts = await deployHelper.deployStateV2('VerifierV2', 'StateV2_deployed', true);
+        state = contracts.state;
+        guWrpr = await deployHelper.deployGenesisUtilsWrapper();
+  });
+
+  it("Check migration", async () => {
+    // 1. init old contract by abi & address
+    const stateContractMigrationHelper = new StateContractMigrationHelper(deployHelper, signers[0]);
+    const oldContractABI = require("../../scripts/StateV2_deployed_abi.json");  // abi of contract that will be upgraded
+    const stateContractAddress = state.address;  // address of contract that will be upgraded
+    const stateContractInstance = await stateContractMigrationHelper.getInitContract({
+        contractNameOrAbi: oldContractABI,
+        address: stateContractAddress,
+    });
+
+    // 2. publish first state
+    const params1 = await publishState(stateContractInstance, stateTransitionsWithProofs[0]);
```

Comment:
Why don't we extend this method (or add another method) in state utils to working with generic state transition?

Reply:
The `publishState` function knows how to work with data input (prepare inputs for transaction) we have on separate json files and we use it in others unit tests. I guess we need to reuse that logic here as well.